### PR TITLE
bench: add more queries to jaeger benches

### PR DESCRIPTION
**Pull request by jacobtrombetta**
# Rationale for this change
The Jaeger benchmarks only measure a small number of queries with a limited amount of column types.. This PR expands the benchmarks to include more queries over most column types.

The `Decimal75` and `TimestampTZ` column types are still not supported in the benchmarks `random_util` module. They are not used in the new queries.

# What changes are included in this PR?
- The `random_util` module is expanded to support most column types.
- Additional queries are added which measure supported SQL functions inside the Jaeger benchmarks.
- The arithmetic query is updated to produce a smaller output table.

# Are these changes tested?
Yes.

### CODE REVIEW

File: crates/proof-of-sql/benches/scaffold/querys.rs [hunk 1]
```diff
@@ -60,6 +60,21 @@ const GROUPBY_COLUMNS: &[(&str, ColumnType, OptionalRandBound)] = &[
     ),
     ("c", ColumnType::Boolean, None),
 ];
+const AGGREGATE_TITLE: &str = "Aggregate";
+const AGGREGATE_SQL: &str = "SELECT SUM(a) FROM table WHERE b = a OR c = 'ab'";
```
Comment (stuarttimwhite) [inline crates/proof-of-sql/benches/scaffold/querys.rs:64]: NIT: Can the string be anything other than "ab", just for readability, given that there are variables a and b? Really not a big deal.

File: crates/proof-of-sql/benches/scaffold/querys.rs [hunk 2]
```diff
@@ -118,6 +118,27 @@ const LARGE_COLUMN_SET_COLUMNS: &[(&str, ColumnType, OptionalRandBound)] = &[
     ("g", ColumnType::VarChar, None),
     ("h", ColumnType::Scalar, None),
 ];
+const COMPLEX_CONDITION_TITLE: &str = "Complex Condition";
+const COMPLEX_CONDITION_SQL: &str =
+    "SELECT * FROM table WHERE (a > c * c AND b < c + 10) OR (d = 'abc')";
```
Comment (stuarttimwhite) [inline crates/proof-of-sql/benches/scaffold/querys.rs:123]: NIT: Same thing as before.

File: crates/proof-of-sql/benches/scaffold/querys.rs (hunk 1 above)
Comment (jacobtrombetta) [reply crates/proof-of-sql/benches/scaffold/querys.rs:64]: The motivation for `ab` and not something like `a` was to limit the size of the result table. Maybe there is a better way to limit the result table size?

File: crates/proof-of-sql/benches/scaffold/querys.rs (hunk 2 above)
Comment (jacobtrombetta) [reply crates/proof-of-sql/benches/scaffold/querys.rs:123]: `abc` follows the same motivation as the last comment.

**Pull request merged**